weak symbols don't work properly in 64-bit Cygwin and should not
be used (even though it will compile with __attribute__((weak)),
it will not linke; see https://trac.sagemath.org/ticket/16152
--- a/src/assertions.hh	2016-07-26 16:21:22.591434100 +0200
+++ b/src/assertions.hh	2016-07-26 16:35:09.500888200 +0200
@@ -98,7 +98,7 @@
 
 namespace Parma_Polyhedra_Library {
 
-#if PPL_CXX_SUPPORTS_ATTRIBUTE_WEAK
+#if PPL_CXX_SUPPORTS_ATTRIBUTE_WEAK && ! (defined(__CYGWIN__) && defined(__x86_64__))
 #define PPL_WEAK_NORETURN __attribute__((weak, noreturn))
 #else
 #define PPL_WEAK_NORETURN __attribute__((noreturn))
